be/src/formats/orc/apache-orc/c++/test/TestColumnReader.cc

```diff
@@ -589,19 +589,31 @@ TEST(TestColumnReader, testSubstructsWithNulls) {
 
     const unsigned char buffer1[] = {0x16, 0x0f};
     EXPECT_CALL(streams, getStreamProxy(1, proto::Stream_Kind_PRESENT, true))
-            .WillRepeatedly(testing::Return(new SeekableArrayInputStream(buffer1, ARRAY_SIZE(buffer1))));
+            // .WillRepeatedly(testing::Return(new SeekableArrayInputStream(buffer1, ARRAY_SIZE(buffer1))));
```

Comment:
add some comments to explain why

Reply:
It's because this code

```
StructColumnReader::StructColumnReader(const Type& type, StripeStreams& stripe) : ColumnReader(type, stripe) {
                    children.push_back(buildReader(child, stripe));
                    fieldIndex.push_back(fi);
                    if (child.getKind() == TypeKind::STRUCT) {
                        // If current child is a struct type, considering it's subfields may need
                        // lazy load, we should add it to lazyLoadChildren too.
                        lazyLoadChildren.push_back(buildReader(child, stripe));
                        lazyLoadFieldIndex.push_back(fi);
}
```

a stream will be used multiple times(and underlying there is unique_ptr and std::move)

So in test case, we have to create input stream every time instead of re-using one. Otherwise, we will get bad memory access.

---

It's quite verbose to explain in code, especially test code. So I think it's to put explanation here.